draft-ietf-trans-gossip.md

```diff
@@ -308,44 +343,116 @@ data SHOULD be sent in an already established TLS session. This makes
 
-HTTPS servers perform a number of sanity checks on feedback from clients
-before storing them:
+Some clients have trust anchors or logs that are locally added (e.g. by an
+administrator or by the user themselves). These additions are
+potentially privacy-sensitive because they can carry information about the
+specific configuration, computer, or user. 
+
+Certificates validated by locally added trust anchors will commonly have no 
+SCTs associated with them, so in this case no action is needed with respect 
+to CT Gossip. SCTs issued by locally added logs MUST NOT be reported via SCT 
+Feedback.
+
+If a certificate is validated by SCTs issued by publicly trusted logs, but 
+chains to a local trust anchor, the client MAY perfom SCT Feedback 
+for this SCT and certificate chain bundle. If it does so, the client MUST
+include the full path of certificates chaining to the local trust anchor in
+the x509\_chain array. Perfoming SCT Feedback in this scenario may be 
+advantageous for the broader Internet and CT ecosystem, but may also disclose 
+information about the client. If the client elects to omit SCT Feedback, it can 
+still choose to perform STH Pollination after fetching an inclusion proof, 
+as specified in  {{#sth-pollination}}.
+
+We require the client to send the full path (or nothing at all) for two 
+reasons. Firstly, it simplifies the operation on the server if there are 
+not two code paths. Secondly, omitting the chain does not actually preserve
+user privacy. The Issuer field in the certificate describes the signing 
+certificate. And if the certificate is being submitted at all, it means the 
+certificate is logged, and has SCTs. This means that the Issuer can be queried
+and obtained from the log - so omitting from the client's submission does
+not actually help user privacy.
 
-  1. if a bit-wise compare of an SCT plus chain matches a pair already
-  in the store, this SCT and chain pair MAY be discarded
+If the HTTPS client has configuration options for not sending cookies
+to third parties, SCTs of third parties MUST be treated as cookies
+with respect to this setting. This prevents third party tracking
+through the use of SCTs/certificates, which would bypass the cookie
+policy.
 
-  1. if the SCT can't be verified to be a valid SCT for the
-  accompanying leaf cert, issued by a known log, the SCT SHOULD be
-  discarded
+\[ TBD: We're thinking about reversing this decision \]
+
+### HTTPS server operation {#feedback-srvop}
+
+HTTPS servers can be configured (or omit configuration), resulting
+in, broadly, two modes of operation. In the simpler mode, the server
+will only track leaf certificates and SCTs applicable to those 
+leaf certificates. In the more complex mode, the server will confirm 
+the client's path validation and store the certificate path. The 
+latter mode requires more configuration, but is necessary to prevent DoS
+attacks on the server's storage space.  
+
```

Comment:
Should we add a note on why simply limiting the size of the data store opens up for a flooding/flushing attack?